props/jetpack_Oli_buttons.h

```diff
@@ -0,0 +1,945 @@
+#define HAVE_DIM        // for EFFECT_POWERSAVE
+#define HAVE_VMBEGIN    // for Begin Volume Menu
+#define HAVE_VMEND      // for End Volume Menu
+#define HAVE_VOLDOWN    // for decrease volume
+#define HAVE_VOLUP      // for increse volume
+#define HAVE_VOLMAX     // for maximum volume reached
+#define HAVE_VOLMIN     // for minimum volume reached
+#endif
+
+#ifdef PROPS_SABER_SABERSENSE_BUTTONS_H
+#define HAVE_BATTERY    // for EFFECT_BATTERY_LEVEL
+#define HAVE_DIM        // for EFFECT_POWERSAVE
+#define HAVE_VMBEGIN    // for Begin Volume Menu
+#define HAVE_VMEND      // for End Volume Menu
+#define HAVE_VOLDOWN    // for decrease volume
+#define HAVE_VOLUP      // for increse volume
+#define HAVE_VOLMAX     // for maximum volume reached
+#define HAVE_VOLMIN     // for minimum volume reached
+#endif
+
+#ifdef PROPS_SABER_SHTOK_BUTTONS_H
+#define HAVE_BATTERY    // for EFFECT_BATTERY_LEVEL
+#define HAVE_DIM        // for EFFECT_POWERSAVE
+#endif
+
+#ifdef PROPS_SABER_SABERTRIO_BUTTONS_H // I know, even them!
+// I have been collecting props from every horizon for testing purposes.
+#define HAVE_BATTERY    // for EFFECT_BATTERY_LEVEL
+#define HAVE_DIM        // for EFFECT_POWERSAVE
+#define HAVE_MUTE       // for Mute Mode
+#define HAVE_VMBEGIN    // for Begin Volume Menu
+#define HAVE_VMEND      // for End Volume Menu
+#define HAVE_VOLDOWN    // for decrease volume
+#define HAVE_VOLUP      // for increase volume
+#define HAVE_VOLMAX     // for maximum volume reached
+#define HAVE_VOLMIN     // for minimum volume reached
+#endif
+
+#ifdef PROPS_BLASTER_BC_BUTTONS_H
+#ifndef HAVE_BATTERY
+#define HAVE_BATTERY    // for EFFECT_BATTERY_LEVEL
+#endif
+#ifndef HAVE_VMBEGIN
+#define HAVE_VMBEGIN    // for Begin Volume Menu
+#endif
+#ifndef HAVE_VMEND
+#define HAVE_VMEND      // for End Volume Menu
+#endif
+#ifndef HAVE_VOLDOWN
+#define HAVE_VOLDOWN    // for decrease volume
+#endif
+#ifndef HAVE_VOLUP
+#define HAVE_VOLUP      // for increse volume
+#endif
+#ifndef HAVE_VOLMAX
+#define HAVE_VOLMAX     // for maximum volume reached
+#endif
+#ifndef HAVE_VOLMIN
+#define HAVE_VOLMIN     // for minimum volume reached
+#endif
+#endif
+
+#ifndef HAVE_BATTERY
+EFFECT(battery);        // for EFFECT_BATTERY_LEVEL
+#endif
+#ifndef HAVE_DIM
+EFFECT(dim);            // for EFFECT_POWERSAVE (dim blades while in idle mode)
+#endif
+#ifndef HAVE_MUTE
+EFFECT(mute);           // Notification before muted ignition to avoid confusion.
+#endif
+#ifndef HAVE_VMBEGIN
+EFFECT(vmbegin);        // for Begin Volume Menu
+#endif
+#ifndef HAVE_VMEND
+EFFECT(vmend);          // for End Volume Menu
+#endif
+#ifndef HAVE_VOLDOWN
+EFFECT(voldown);        // for decrease volume
+#endif
+#ifndef HAVE_VOLUP
+EFFECT(volup);          // for increase volume
+#endif
+#ifndef HAVE_VOLMAX
+EFFECT(volmax);         // for maximum volume reached
+#endif
+#ifndef HAVE_VOLMIN
+EFFECT(volmin);         // for minimum volume reached
+#endif
+
+#ifdef HAVE_BATTERY

```

Comment:
I think it's fine to #undef something that isn't defined, so the #ifdef isn't needed I think.